src/main/java/frc/robot/subsystems/color/SignalingSubsystem.java

```diff
@@ -0,0 +1,244 @@
+package frc.robot.subsystems.color;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.color.patterns.BIFlag;
+import frc.robot.subsystems.color.patterns.FireFlyPattern;
+import frc.robot.subsystems.color.patterns.PatternNode;
+import frc.robot.subsystems.color.patterns.RainbowPattern;
+import frc.robot.subsystems.color.patterns.TransFlag;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+public class SignalingSubsystem extends SubsystemBase {
+  private static final int patternUpdateFrequency = 10;
+
+  private final AddressableLED ledStrip;
+  private final AddressableLEDBuffer ledBuffer;
+  
+  private static final int numberOfLEDS = Constants.LED_COUNT;
+
+  private int tick;
+  private int patternTick;
+  private final Timer amplifierTimer;
+  private final Timer rumbleTimer;
+  private double rumbleEndTime = 0;
+  private boolean isAllianceAmplified;
+  private boolean isOpponentAmplified;
+
+  private final boolean isRedAlliance;
+
+  private DisablePattern disablePattern = DisablePattern.getRandom();
+
+  private final Consumer<Double> driverRumbleConsumer;
+
+  public SignalingSubsystem(int ID, Consumer<Double> driverRumbleConsumer) {
+    this.driverRumbleConsumer = driverRumbleConsumer;
+
+    tick = 0;
+    patternTick = 0;
+    amplifierTimer = new Timer();
+    rumbleTimer = new Timer();
+    isAllianceAmplified = false;
+    isOpponentAmplified = false;
+
+    this.isRedAlliance = DriverStation.getAlliance().equals(DriverStation.Alliance.Red);
+
+    // Initialize LED Strip
+    ledStrip = new AddressableLED(ID);
+    ledBuffer = new AddressableLEDBuffer(numberOfLEDS);
+    ledStrip.setLength(numberOfLEDS);
+    ledStrip.setData(ledBuffer);
+    ledStrip.start();
+
+    // Setup Disable Patterns
+    TransFlag.numberOfLEDS = numberOfLEDS;
+    FireFlyPattern.numberOfLEDS = numberOfLEDS;
+    BIFlag.numberOfLEDS = numberOfLEDS;
+    RainbowPattern.numberOfLEDS = numberOfLEDS;
```

Comment:
Is there a good reason not to just have the patterns use the Constants.LED_COUNT directly? This is an unusual way to set values of another class.